sashimi/state.py

```diff
@@ -569,6 +572,41 @@ def end_experiment(self):
         self.experiment_start_event.clear()
         self.saver.save_queue.clear()
         self.send_scansave_settings()
+        self.current_exp_state = GlobalState.PAUSED
+
+    def is_exp_started(self) -> bool:
+        """
+        check if the experiment has started:
+        looks for tha change in the value hold by current_exp_running
+
+        Returns:
+            bool
+        """
+        if (
+            self.current_exp_state == GlobalState.EXPERIMENT_RUNNING
+            and self.prev_exp_state == GlobalState.PAUSED
+        ):
+            self.prev_exp_state = GlobalState.EXPERIMENT_RUNNING
+            return True
+        else:
+            return False
+
+    def is_exp_ended(self) -> bool:
+        """
+        check if the experiment has ended:
+        looks for tha change in the value hold by current_exp_running
+
+        Returns:
+            bool
+        """
+        if (
+            self.prev_exp_state == GlobalState.EXPERIMENT_RUNNING
+            and self.current_exp_state == GlobalState.PAUSED
+        ):
+            self.prev_exp_state = GlobalState.PAUSED
+            return True
+        else:
+            return False
```

Comment:
Moved the logic inside state.py

here a two functions check whether the experiment started/ended and return a bool